ExampleMod/Content/Tiles/ExampleCrystal.cs

```diff
@@ -0,0 +1,118 @@
+using Terraria.ObjectData;
+
+namespace ExampleMod.Content.Tiles
+{
+    internal class ExampleCrystal : ModTile
+    {
+        public override void SetStaticDefaults()
+        {
+            TileObjectData.newTile.Width = 1;
+            TileObjectData.newTile.Height = 1;
+            TileObjectData.newTile.CoordinateWidth = 16; // Width of tile on spritesheet
+            TileObjectData.newTile.CoordinateHeights = new int[] { 16 }; // Sets an offset
+            TileObjectData.newTile.CoordinatePadding = 2; // Sets an "empty space" that should be between each 16x16 sprite in spritesheet
+            TileObjectData.newTile.StyleHorizontal = true;
+            TileObjectData.addTile(Type);
+            Main.tileFrameImportant[Type] = true;
+            Main.tileObsidianKill[Type] = true;
+            Main.tileShine2[Type] = true;
+            Main.tileShine[Type] = 500;
+            Main.tileSpelunker[Type] = true;
+        }
+
+        public override ushort GetMapOption(int i, int j) => (ushort)(Main.tile[i, j].TileFrameX / 18); // this override is redundant, but in case if you want more map names option ig, then you can leave it alone?
+
+        public override void KillTile(int i, int j, ref bool fail, ref bool effectOnly, ref bool noItem)
+        {
+            int toDrop = 0;
+			if (!fail)
+			{
+				toDrop = ModContent.ItemType<ExampleCrystalItem>();
+				DustType = ModContent.DustType<Sparkle>();
+			}
+			if (toDrop > 0) Item.NewItem(Entity.GetSource_None(), i * 16, j * 16, 16, 16, toDrop);
+        }
+
+        public override bool TileFrame(int i, int j, ref bool resetFrame, ref bool noBreak)
+        {
+            Main.tile[i, j].TileType = TileID.Crystals;
+            WorldGen.TileFrame(i, j, resetFrame, noBreak); // applies vanilla Crystal Shards behavior
+            Main.tile[i, j].TileType = Type;
+            return false;
+        }
+    }
+    // this class does some spreading
+    internal class ExampleCrystalGrow : GlobalTile
+    {
+        public override void RandomUpdate(int i, int j, int type)
+        {
+            if (Main.hardMode && type == TileID.Stone && j > Main.rockLayer && WorldGen.genRand.NextBool(110))
+                // this code line above means, that your crystal will grow only if: it's Hardmode, on Stone, in rock layer, and with same chance as vanilla Crystal Shards.
+            {
+                int coordinateChance = WorldGen.genRand.Next(4);
+                int additionalX = 0;
+                int additionalY = 0;
+                if (coordinateChance == 0)
+                {
+                    additionalX = -1;
+                }
+                else if (coordinateChance == 1)
+                {
+                    additionalX = 1;
+                }
+                else if (coordinateChance == 0)
+                {
+                    additionalY = -1;
+                }
+                else
+                {
+                    additionalY = 1;
+                }
```

Comment:
Merge these ifs
```
                if (coordinateChance == 0)
                {
                    additionalX = -1;
                    additionalY = -1;
                }
                else if (coordinateChance == 1)
                {
                    additionalX = 1;
                    additionalY = 1;
                }
```

Alternatively even
```
additionalX = !coordinateChance ? -1 : 1;
additionalY = additionalX;
```
But I guess that's too advanced for example mod

Reply:
trueee
alr, will commit another change now

Reply:
and wait, isn't `coordinateChance` an integer?
so you can't use `!coordinateChance` at this point

Reply:
Yes it's an integer, but booleans are integers too, really. False = 0, True = 1
So change it to 
```
additionalX = coordinateChance != 0 ? 1 : -1;
additionalY = additionalX;
```
and it would work (otherwise you would have to convert to boolean)

Reply:
trueeee, but `coordinateChance` is `WorldGen.genRand.Next(4)`, and `additional(X/Y)` default values are 0, so there should be a check for if `coordinateCheck` is not 2, like:
```
if (coordinateCheck != 2)
{
// code
}
```

Reply:
yeah, whatever